pkg/pool-utils/contracts/controllers/AssetManagedLBPController.sol

```diff
@@ -0,0 +1,158 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/ERC20Helpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+import "../interfaces/IControlledLiquidityBootstrappingPool.sol";
+import "./BasePoolController.sol";
+
+/**
+ * @notice Controller for an unseeded LiquidityBootstrappingPool. It also serves as the asset manager.
+ * Using the `UnseededLiquidityBootstrappingPoolFactory` will deploy this controller and a pool.
+ * After the contracts are deployed by the factory, the manager can call `fundPool` on the controller
+ * with the initial balances. This funds the pool using the manager's pool tokens. The reserve tokens
+ * will be supplied using an AAVE flashloan, then immediately withdrawn using the asset manager
+ * functionality to repay the loan.
+ *
+ * The result is an apparently fully funded pool, but with a zero cash balance in the reserve token.
+ * This means project tokens cannot be "sold back" into the pool - at least until enough are sold to
+ * build up the real balance.
+ *
+ * Note that there is a flash loan fee, so the manager will need enough reserve tokens to pay it (and
+ * must approve the controller to pull them during the `fundPool` call). The amount of the fee can be
+ * calculated using the `getFlashLoanFeePercentage` function on the controller.
+ *
+ * The controller can then be used by the manager for regular LBP functions.
+ */
+contract AssetManagedLBPController is BasePoolController, IControlledLiquidityBootstrappingPool {
+    using FixedPoint for uint256;
+
+    // WeightedPoolUserData type - duplicating here to avoid a circular dependency
+    enum JoinKind { INIT }
+
+    IVault private immutable _vault;
+    IERC20 private immutable _reserveToken;
+    uint256 private immutable _projectTokenIndex;
+    uint256 private immutable _reserveTokenIndex;
+    uint256 private _virtualReserveAmount;
+
+    constructor(
+        BasePoolRights memory baseRights,
+        IVault vault,
+        IERC20 reserveToken,
+        bool projectTokenFirst,
+        address manager
+    ) BasePoolController(encodePermissions(baseRights), manager) {
+        _reserveToken = reserveToken;
+        _projectTokenIndex = projectTokenFirst ? 0 : 1;
+        _reserveTokenIndex = projectTokenFirst ? 1 : 0;
+        _vault = vault;
+    }
+
+    /**
+     * @dev Call this function instead of the usual `joinPool` (init). It will borrow the necessary amount
+     * of reserve tokens using a DODO flashloan.
+     */
+    function fundPool(uint256[] memory initialBalances) external onlyManager {
```

Comment:
In the case of using fundPool instead of `joinPool`, how are the parameters we used to be passed to join pool process since fundPool only takes the initialBalances?
ex:
```
        bytes memory userData = abi.encode(0, poolConfig.amounts); // JOIN_KIND_INIT = 0
        Vault(VAULT).joinPool(
            poolId,
            address(this), // sender
            address(this), // recipient
            Vault.JoinPoolRequest(poolConfig.tokens, poolConfig.amounts, userData, false)
        );
      ```

Reply:
No other arguments are required. It's doing an initialization join, which only requires the initial balances. In the internal join call on the Vault, the sender is the controller contract, and the recipient is the manager. It pulls project tokens from - and sends BPT to - the manager.